### ISSUE TITLE
Settings `processors` via environment property doesn't work

### ISSUE THREAD

**Issue by jloisel** [elastic/elasticsearch#33633]
**Elasticsearch version** (`bin/elasticsearch --version`): 6.3.2

**Plugins installed**: []

**JVM version** (`java -version`): built-in from docker image `elasticsearch-oss`

**OS version** (`uname -a` if on a Unix-like system): Ubuntu 18.04LTS

**Description of the problem including expected versus actual behavior**:

Settings `processors` property works when set in elasticsearch.yml, but has no effect when passed a docker environment property.

**Steps to reproduce**:

Run elasticsearch with docker and try to pass -E "processors=1" for example. It doesn't have any effect on the node allocatedProcessors.

**Comment by jloisel** [comment]
Created my own `elasticsearch.yml` as a workaround and shared it via NFS with all the cluster nodes. Still interested in setting this via environment variable as it's less painful to manage.

**Comment by jloisel** [comment]
As a side effect, I'm running an Elasticsearch cluster on Rancher 1.6.x and have had issues with the number of CPUs being recognized by Elasticsearch. It was set to 1.

I configured the issue is not elasticsearch in itself, but rather the number of CPUs being detected in the docker container by JDK10:
```
jshell> Runtime.getRuntime().availableProcessors()
$1 ==> 1
```
Note that this is not happening when starting the container manually through command-line or through docker-compose. This must be something Rancher messes up.

**Comment by jasontedor** [comment]
I opened elastic/elasticsearch-docker#195.

**Comment by jasontedor** [comment]
Closed by elastic/elasticsearch-docker#195

**Comment by jasontedor** [comment]
This will be fixed in the 6.4.1 image.

**Comment by jloisel** [comment]
Thanks!
